### PR TITLE
fix build issue for Linux and Mac

### PR DESCRIPTION
**Pull request by xrdavies**
fix build issue for Linux and Mac
move log_init and rpc_init after dnet fork a new process

### CODE REVIEW

File: client/init.c [hunk 1]
```diff
@@ -70,9 +70,6 @@ int xdag_init(int argc, char **argv, int isGui)
 	if (!isGui) {
 		printf("%s client/server, version %s.\n", g_progname, XDAG_VERSION);
 	}
-	
-	/* initialize log system */
-	if (xdag_log_init()) return -1;
```
Comment (jonano614) [inline client/init.c]: I don't think it is good idea to move initialization of log system to the end of initialization

File: client/pool.c [hunk 2]
```diff
@@ -1215,6 +1215,7 @@ static void print_connection(FILE *out, int index, struct connection_pool_data *
 		connection_state_to_string(conn_data->state), ip_port_str, in_out_str, connection_calculate_unpaid_shares(conn_data));
 }
 
+int print_connections(FILE *out);
```
Comment (jonano614) [inline client/pool.c]: Do we need this declaration?

File: client/init.c (hunk 1 above)
Comment (xrdavies) [reply client/init.c]: It's original  here. I moved it to the beginning last time. And I found xdag_transport_start will fork a process from time to time then the logging system is not working.

File: client/pool.c (hunk 2 above)
Comment (xrdavies) [reply client/pool.c]: Yes, please find the build error from AnotherYou. It's 
"
pool.c:1218:5: warning: no previous prototype for ‘print_connections’ [-Wmissing                                                                             -prototypes]
 int print_connections(FILE *out)
"

File: client/init.c (hunk 1 above)
Comment (jonano614) [reply client/init.c]: Ok, I see.

**Pull request merged**